src/test/java/com/hotels/bdp/circustrain/bigquery/table/service/partitioned/PartitionedTableServiceTest.java

```diff
@@ -74,7 +75,7 @@ public void getTable() {
   public void getPartitions() {
     when(tableResult.iterateAll()).thenReturn(rows);
     List<Partition> result = partitionedTableService.getPartitions();
-    assertThat(result, is(factory.generate(eq(partitionBy), eq(rows))));
+    assertThat(result, is(factory.generate(eq(partitionBy), eq(type), eq(rows))));
```

Comment:
These test aren't good, you are asserting that: null, is(null). Must have slipped past in previous PR's. 
You are not testing any of the arguments try changing the test: `assertThat(result, is(factory.generate(eq("foo"), eq("bar"), eq(rows))));` it will still pass green :)
Rewrite them please, mock the factory call properly will help.

Also please rename `factory` -> `partitionGenerator`.